org.eclipse.xtext.testing/src/org/eclipse/xtext/testing/extensions/InjectionExtension.java

```diff
@@ -59,10 +59,14 @@ public void beforeEach(ExtensionContext context) throws Exception {
 			Injector injector = injectorProvider.getInjector();
 			if (injector != null) {
 				Object testInstance = context.getRequiredTestInstance();
-				TestInstances requiredTestInstances = context.getRequiredTestInstances();
 				injector.injectMembers(testInstance);
-				for (Object o : requiredTestInstances.getEnclosingInstances()) {
-					injector.injectMembers(o);
+				try {
+					TestInstances requiredTestInstances = context.getRequiredTestInstances();
+					for (Object o : requiredTestInstances.getEnclosingInstances()) {
+						injector.injectMembers(o);
+					}
+				} catch (NoSuchMethodError e) {
+					// OK, getRequiredTestInstances is not there in Junit5 < 5.4
```

Comment:
Should we throw an `ExtensionConfigurationException` if the `testInstance` is an instance of a non-static inner class?

Reply:
ähm. this is all about non static inner classes

Reply:
I mean from within the catch block.

Reply:
updated the PR